### DISCUSSION TITLE
⭐ 스터디 2기 참여자 후기 취합 ⭐

### DISCUSSION THREAD

**Discussion by DaleSeo** [DaleStudy/leetcode-study#601] [공지사항]
스터디 2기 참여자 여러분, 안녕하세요.

우리 스터디를 시작한지 엊그제 같은데 어느덧 15주가 흘러 이제 마지막 주차에 접어들었습니다. 매주 열심히 문제를 푸시고 정기 모임에도 적극적으로 참석해주셔서 재미있게 스터디가 진행된 것 같습니다. 끝까지 완주해주신 약 10분께 특히 더 고생 많으셨고 축하한다는 말씀드리고 싶어요! 👏👏👏

제가 첫 모임에서도 말씀드렸던 것처럼 Blind 75 문제를 한 번 다 푸셨다고 해서 갑자기 많은 게 달라지지는 않을 것입니다. 여전히 실전 코딩 테스트와 영어 기술 면접에서는 갑자기 머리가 하얘지고 막상 푸신 문제도 나와도 까먹으셔서 좌절하시는 경험을 하시게 될 거에요. 하지만 시작이 반이라고 이렇게 한 번 쭉 풀어본 것만으라도 앞으로 개발자로 해외 취업을 준비하시는데 큰 자양분이 되실거라 생각합니다.

스터디 2기는 이번 주를 끝으로 종료하지만 저희 디스코드 커뮤니티는 언제나 열려있으니 스터디가 끝나고도 종종 놀러오셔서 좋은 교류를 나누었으면 좋겠습니다. 그리고 혹시라도 나중에라도 해외 취업에 성공하셨다면 꼭꼭 스토리를 공유해주셔서 다른 분들에게 영감과 용기를 나누어 주셨으면 좋겠습니다.

마지막으로 여러분의 생생한 스터디 참여 경험을 공유할 수 있는 후기를 좀 부탁드리려고 해요! 이 토론에 댓글로 후기를 달아주시면 @DaleStudy/website 개발팀이 웹사이트에 실어 주실 것입니다. 😁

스터디 1기 분들의 후기는 아래 웹사이트 캡쳐를 참고하세요.
![Shot 2024-11-17 at 11 39 00@2x](https://github.com/user-attachments/assets/403b38f2-7c2e-4f3b-a945-1273fda15c5d)

다들 수고 많으셨습니다! 마지막 모임에서 뵈요!

**Comment by obzva** [comment]
어떤 주는 5일 동안 하루에 한 문제씩, 또 다른 주는 하루만에 몰아서 다섯 문제를. 그렇게 조금씩 알고리즘 문제를 푼지 벌써 15주나 지났다는게 믿기지 않습니다. 
비슷한 목표를 가진 사람들과 함께이기에 꾸준히 문제풀이를 해낼 수 있었고, 다른 분들께 조금이라도 도움이 되었으면 좋겠다는 생각으로 풀이를 적다보니 코드도 좀 더 꼼꼼하게 쓸 수 있었습니다. 
혼자 문제 푸는 것이 어려우신 분, 본인의 생각을 남들에게 전달하는게 익숙하지 않은 분, 성숙하게 피드백을 주고받는 연습이 필요하신 분들께 이 스터디를 강력 추천합니다.

**Comment by HC-kang** [comment]
15주가 정말 순식간에 지나갔네요. 직장 생활과 병행하면서 알고리즘 풀이를 하려니 정말 힘들긴 했지만 그만큼 보람찹니다!

제가 생각하는 이 스터디의 장점은 두가지 인데요,
첫째는 '자율적인 꾸준함'을 만들기 좋았다는 점입니다. 강제성이 있는것은 아니다보니 심적인 부담감이 덜하지만, 따라갈 수 있는 명확한 목표가 제시된다는 점이 바쁜 와중에도 따라갈 좋은 동기가 부여 되었습니다. 
둘째로는, 단순히 번호순 혹은 난이도순 풀이가 아니라 비슷한 카테고리의 알고리즘을 순차적으로 접할 수 있도록 편성된 달레님의 문제 큐레이션이 굉장히 좋았습니다. 아마 함께하신 분들도 참여하는 과정에서 모두가 느끼셨을텐데, '어? 이거 지난 주 문제의 심화버전인것 같네?' 하면서 복습과 심화과정을 함께 다룰 수 있었습니다.

그리고 또하나의 특장점은, 상호 리뷰하고 풀이를 공유하는 과정에서 다른분들의 참여와 풀이를 보면서 또 자연스러운 동기부여가 되었습니다.
특히 몇몇 분들은 거의 기술 아티클 수준으로 뇌리에 박히는 풀이를 작성해주셔서 아마 후에 참여하시는 분들은 과거 풀이 기록들을 찾아보시는것도 도움이 많이 될 것 같네요.

다음에도 이런 기회가 또 있다면 꼭 다시 참여하고싶습니다. 알고리즘 문제 풀이의 꾸준함이 필요하신 분들, 꼭 해외 취업을 준비하시는 분들이 아니더라도 이 스터디를 추천드립니다.

**Comment by EgonD3V** [comment]
다른 알고리즘 스터디와 달리 인터뷰를 가정하고 코드를 작성했던 점이 가장 달랐던 것 같습니다. 

만약 내가 인터뷰어라면 어떤 점을 물어볼지, 내가 인터뷰이로서 어떤 식으로 대답을 해야할지 시뮬레이션을 해볼 수 있어 좋았습니다. 그러다보니 어떻게 변수와 로직을 구현해야 논리가 명확할지, 이 논리를 내가 이해하고 상대를 설득하려면 어떤 식으로 설명해야 할지를 15주 동안 고민한 경험을 할 수 있었다고 말씀드리고 싶습니다.

이전에도 알고리즘 문제들을 풀어보기는 했지만, 단순히 알고리즘 온라인 저지 시스템 사이트에서 Pass or Fail만 확인 했었는데, 시간복잡도와 공간복잡도를 문제마다 계산해보고 내가 구상한 알고리즘을 다른 사람이 볼 때 어떨지를 고려하며 문제를 푸는 경험을 통해 많이 배울 수 있었습니다. 또한 같은 문제를 가능한 여러 방법으로 풀어보고 제약 조건을 추가해보기도 하며 고민해보는 경험도 좋았습니다.

15주간의 스터디를 마치며, Blind 75라는 좋은 문제집을 각 문제별 연계와 확장을 고려해서 주차 별로 배치하여 여러 사람들과 함께 공부할 수 있는 자리를 마련해주신 Dale님께 감사를 표합니다.

**Comment by jaejeong1** [comment]
먼저 15주 간 스터디를 이끌어 주신 @DaleSeo 님과 @SamTheKorean 님 및 코치분들께 감사인사를 드립니다.
덕분에 혼자서 하기 어려웠던 꾸준한 알고리즘 스터디를 완주할 수 있었습니다. 

혼자서 공부할때 느꼈던 막혔을 때의 어려움을 풀 수 있는 수단으로서 큰 도움이 됐습니다. 다른 스터디원들의 풀이 또는 달레님의 해설은 빠른 트러블 슈팅과 알고리즘 풀이 시야를 빠르게 넓히는데 효과적이었습니다.
또한, 매주 온라인 스터디를 진행하는데 이때 풀이를 설명하는 경험이 이해의 깊이를 늘리는데 효과적이었고, 실제 인터뷰 과정과도 유사해 추후 커리어 확장에도 긍정적인 영향을 기대할 수 있었습니다.
그리고 간간히 여러 나라와 근무 환경에 있는 스터디원들과의 경험/근황 공유 및 스몰 토크도 유익하고 즐거웠습니다.

앞으로도 이렇게 유익한 해외취업 알고리즘 스터디로 긍정적인 영향을 받은 분들이 더 많아지면 좋겠습니다.
감사합니다!

**Comment by TonyKim9401** [comment]
앗 금방 달은 후에 말씀 드리겠습니다.

**Comment by Sunjae95** [comment]
저는 코딩테스트를 잘 알지는 못하지만 그렇다고 모르지 않는 수준의 개발자였어요. 매번 공부하다가 중간에 포기하고 매번 스택, 큐, bfs정도의 레벨까지만 풀고 dp같이 어려운 문제를 만나면 다음에 해야지 하는 날이 많았어요. 중간에 포기하는 사람 같아서 이번에 달레스터디를 통해 1문제씩이라도 15주 동안 완주해 보는 사람이 되고자 스터디에 참여했습니다. 
달레스터디를 꾸준히 할 수 있었던 건 구성이 좋은 커리큘럼과 스터디원들과 소통하는 점이 제가 완주할 수 있었던 이유였어요. 매주 easy 난이도부터 medium 많게는 hard까지 풀 수 있었는데, 유형이 비슷한 문제가 다음 주에도 나오는 경우도 있어 다시 한번 회고하며 머릿속에 영구저장되는 경험을 했어요. 그리고 스터디원들과 대화를 하면서 내가 몰랐던 부분, 다른 사람의 접근 방식을 간접적으로 들으면서 문제에 대한 접근 방식을 알게 되어 문제 울렁증을 극복할 수 있었어요.
만약 달레스터디를 고민하시는 분이 계신다면 무조건 추천하며, 준비물은 스터디원들과 적극적으로 소통하는 마음가짐을 가지고 오는 것을 추천합니다!

-리뷰 안올려도 되는부분-
15주동안 함께 해주신 달레님, 코치분들, 스터디원분들 너무 감사합니다. 많은 경험을 통해 성장할 수 있었어요 :)
